simulators/eth2/engine/running_testnet.go

```diff
@@ -126,12 +199,14 @@ func (t *Testnet) WaitForFinality(ctx context.Context) (common.Checkpoint, error
 					}
 
 					var checkpoints eth2api.FinalityCheckpoints
-					if exists, err := beaconapi.FinalityCheckpoints(ctx, b.API, eth2api.StateIdRoot(headInfo.Header.Message.StateRoot), &checkpoints); err != nil {
-						ch <- res{err: fmt.Errorf("beacon %d: failed to poll finality checkpoint: %v", i, err)}
-						return
-					} else if !exists {
-						ch <- res{err: fmt.Errorf("beacon %d: Expected state for head block", i)}
-						return
+					if exists, err := beaconapi.FinalityCheckpoints(ctx, b.API, eth2api.StateIdRoot(headInfo.Header.Message.StateRoot), &checkpoints); err != nil || !exists {
```

Comment:
Could we add these changes and few blocks below to `simulators/eth2/testnet/running_testnet.go` so we could pass `eth2/testnet` too? I don't want to just copy and make PR from it but if you want, I can

Reply:
I think it would be better to create another PR to do this to not mix different simulators on the same PR.